### PR TITLE
fix: Trial metric chart series always have unique colors

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

On the experiment Metrics tab, we put metrics with similar names on the same plot. Ideally training loss and validation loss would be plotted together.
In the example, loss and val_loss were plotted together but both coded in the training metric group.
~Under this change, the metric group coloring will look for prefixes, and "val_" will override the metric being in training metric group.~

**Update: Removes blue/orange coloring for training/validation special groups, instead using glasbey colors; this avoids issues with multiple series being resolved to the same chart + training group**

## Test Plan

Experiment `/det/experiments/5518/metrics` displays validation_loss and loss on the same chart

Update `/det/experiments/5522/metrics` shows multiple lines in one chart

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by garrett361 — APPROVED** [review]
Looks good as a temporary fix, but longer-term IMO it makes more sense to group and color based on the new metric groups and exact metric name matches, rather than based on special prefix-matches for only some groups.

**Review by johnkim-det — APPROVED** [review]
LGTM
